### ISSUE TITLE
Implement mDNS announcer

### ISSUE THREAD

**Issue by Timvrakas** [stanford-ssi/SpaceSalmon#45]
There's a few things I want to think about
- UDP stream to any IP address based on a TCP request. Need to stop and restart UDP loop when a new request comes in. We had this working with Flynn's 140e demo.
- DHCP
- mDNS
- Some sort of MAC address derivation scheme

**Comment by mikh3x4** [comment]
First point is now implemented

**Comment by mikh3x4** [comment]
I’m also slightly worried about dhcp giving the ip for launch critical components

**Comment by Timvrakas** [comment]
Yeah I agree. What I landed on was having the first section of the subnet (like 0-32) be reserved for static devices, and the rest allocated with DHCP. In particular
- quail
- telemetry proxy server (the backend)
- cameras

Were all static.

I think MDNS could be cool. You can avoid IP addresses entirely (just use auto-generated link-local addresses) and have quail-etc advertise "I have a TCP telemetry server on X port" and then the backend can just scan the network and connect to all devices. If you had multiple embedded nodes and wanted to be able to reconfigure without re-flashing, that's how you would handle it I think.

The security cameras already advertise themselves over mDNS!
